packages/mobile/src/hooks/useLoadOfflineTracks.ts

```diff
@@ -153,12 +153,19 @@ export const useOfflineCollectionLineup = (
   })
 
   const fetchLocalContent = useCallback(() => {
+    const collectionTrackIds = new Set(
+      collection?.playlist_contents?.track_ids?.map(
+        (trackData) => trackData.track
+      ) ?? []
+    )
+
     if (isOfflineModeEnabled && collectionId) {
       const lineupTracks = Object.values(offlineTracks)
-        .filter((track) =>
-          track.offline?.reasons_for_download.some(
-            (reason) => reason.collection_id === collectionId.toString()
-          )
+        .filter(
+          (track) =>
+            track.offline?.reasons_for_download.some(
+              (reason) => reason.collection_id === collectionId.toString()
+            ) || collectionTrackIds.has(track.track_id)
```

Comment:
reasoning: a track in a collection may be downloaded from the favorited tracks page, but not yet have the collection added as a second reason for download (since the download queue hasn't gotten there yet).

Reply:
sadly this doesnt work the other way around- we only know to include a downloaded track in the favorited tracks tab if it has the favorited tracks collection as a reason for download. cant fix this until we have a way of keeping the full list of favorited track metadata before they are all downloaded - i.e. can do what we do here on line 156 but for the favorited tracks "collection".

Reply:
Great call. Sad we don't have a track list for favorites yet.